src/Overlay.js

```diff
@@ -120,12 +130,73 @@ function wrapRefs(props, arrowProps) {
     aRef.__wrapped || (aRef.__wrapped = (r) => aRef(findDOMNode(r)));
 }
 
-function Overlay({ children: overlay, transition, ...outerProps }) {
+const marginModifiers = [
+  {
+    name: 'marginOffsets',
+    enabled: true,
+    phase: 'beforeRead',
+    effect: ({ state }) => {
+      return () => {
+        state.elements.reference.removeAttribute('aria-describedby');
+      };
+    },
+    fn: ({ state, name }) => {
+      const { popper: popperRect } = state.rects;
+      const { popper, arrow } = state.elements;
+
+      const popperMargins = getMargins(popper);
+      const arrowMargins = arrow && getMargins(arrow);
+
+      state.rects.popper = {
+        ...popperRect,
+        width: popperRect.width + popperMargins.left + popperMargins.right,
+        height: popperRect.height + popperMargins.top + popperMargins.bottom,
+        y: popperRect.y - popperMargins.top,
+        x: popperRect.x - popperMargins.left,
+      };
+
+      state.modifiersData[name] = {
+        popper: popperMargins,
+        arrow: arrowMargins,
+      };
+    },
+  },
+  {
+    name: 'marginOffsetArrow',
+    enabled: true,
+    phase: 'main',
+    requiresIfExists: ['arrow'],
+    fn({ state }) {
+      if (!state.elements.arrow) return;
+
+      const { arrow: arrowMargin } = state.modifiersData.marginOffsets;
+
+      const offsets = state.modifiersData.arrow;
+
+      if (offsets.x != null) offsets.x -= arrowMargin.left;
+      if (offsets.y != null) offsets.y -= arrowMargin.top;
```

Comment:
this is not strictly correct because it assumes margin on either side is the same, but that is true for bootstrap. I think anyone modifying styles should do it correctly via popper

Reply:
Should we add anything to the docs about this?

Reply:
might want to add a comment here, at least